packages/exec/src/toolrunner.ts

```diff
@@ -524,6 +524,15 @@ export class ToolRunner extends events.EventEmitter {
           resolve(exitCode)
         }
       })
+
+      if (this.options.stdin) {
+        if (!cp.stdin) {
+          throw new Error('child process missing stdin')
+        }
+
+        cp.stdin.setDefaultEncoding('utf-8')
```

Comment:
Does `setDefaultEncoding` need to be called? I would think the buffer would just be written verbatim. Or does calling `setDefaultEncoding` cause a BOM to be written first?

Reply:
Honest answer: every single example calls `setDefaultEncoding`, and the documentation for the behavior without seems undefined. My best guess is that it falls back to the OS encoding, which can different between OSes (Windows is UTF-16 IIRC).

As far as I can tell, it doesn't send a BOM. I'm open to changing it, because I admittedly don't understand it.

Reply:
Thanks! Makes sense a BOM would not be sent. I think BOMs are typically used for file streams only, not stdin/stdout.

Based on the docs for [`stream.Writable.write()`](https://nodejs.org/api/stream.html#stream_writable_write_chunk_encoding_callback) my interpretation is `setDefaultEncoding` is only useful when `write` is called with a string, and encoding is not specified.

In our case, the caller is always providing a Buffer (raw bytes). I think it's safe to omit setting the default encoding. Omitting the line shouldnt change behavior, but might avoid confusion.

Reply:
Done